core/java/iesi-core/src/main/java/io/metadew/iesi/metadata/configuration/script/ScriptVersionConfiguration.java

```diff
@@ -173,26 +225,40 @@ private String updateStatement(ScriptVersion scriptVersion){
                 " AND SCRIPT_VRS_NB = " + SQLTools.getStringForSQL(scriptVersion.getMetadataKey().getScriptKey().getScriptVersion()) + ";";
     }
 
-    private String deleteStatement(ScriptVersionKey scriptVersionKey) {
-        return "DELETE FROM " + getMetadataRepository().getTableNameByLabel("ScriptVersions") +
-                " WHERE SCRIPT_ID = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptId()) +
-                " AND SCRIPT_VRS_NB = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptVersion()) + ";";
+    public boolean exists(ScriptVersionKey scriptVersionKey) {
+        String query = "select SCRIPT_ID, SCRIPT_VRS_NB, SCRIPT_VRS_DSC from " + getMetadataRepository().getTableNameByLabel("ScriptVersions")
+                + " where SCRIPT_ID = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptId()) +
+                " and SCRIPT_VRS_NB = " + SQLTools.getStringForSQL(scriptVersionKey.getScriptKey().getScriptVersion()) +
+                " and DELETED_AT = 'NA' ;";
```

Comment:
Doesn't ScriptVersionKey contain deletedAt? don't e need to use this value here?